dashboard/app/models/census/census_hoc2017v1.rb

```diff
@@ -0,0 +1,44 @@
+# == Schema Information
+#
+# Table name: census_submissions
+#
+#  id                           :integer          not null, primary key
+#  type                         :string(255)      not null
+#  submitter_email_address      :string(255)
+#  submitter_name               :string(255)
+#  submitter_role               :string(255)
+#  school_year                  :integer          not null
+#  how_many_do_hoc              :string(255)
+#  how_many_after_school        :string(255)
+#  how_many_10_hours            :string(255)
+#  how_many_20_hours            :string(255)
+#  other_classes_under_20_hours :boolean
+#  topic_blocks                 :boolean
+#  topic_text                   :boolean
+#  topic_robots                 :boolean
+#  topic_internet               :boolean
+#  topic_security               :boolean
+#  topic_data                   :boolean
+#  topic_web_design             :boolean
+#  topic_game_design            :boolean
+#  topic_other                  :boolean
+#  topic_other_description      :string(255)
+#  topic_do_not_know            :boolean
+#  class_frequency              :string(255)
+#  tell_us_more                 :string(255)
+#  pledged                      :boolean
+#  created_at                   :datetime         not null
+#  updated_at                   :datetime         not null
+#
+# Indexes
+#
+#  index_census_submissions_on_school_year_and_id  (school_year,id)
+#
+
+# This class represents submissions from the original census questions on
+# the Hour of Code signup page.
+#
+class Census::CensusHoc2017v1 < Census::CensusSubmission
```

Comment:
Would we ever want to migrate older records to newer versions to normalize our data (making some of these layers redundant) or will we keep our record of submissions static and do migration/rollup entirely on the reporting side, later?  If we would want to migrate, is there a strategy for that with this inheritance-versioned structure?

Reply:
I don't expect to upgrade older records. The idea is to store the responses as they were received. Eventually we are going to have a census_summaries table that will summarize all of the data about a given school, whether from multiple survey submissions or from alternative data sources (like state curriculum info.) The logic to populate that table will need to understand how to interpret the different versions and that is were we would likely have any logic to treat an older record like a newer one.